components/breadcrumbs/breadcrumb.js

```diff
@@ -0,0 +1,89 @@
+import '../icons/icon.js';
+import '../link/link.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class BreadCrumb extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			compact: {
+				type: Boolean,
+				reflect: true
+			},
+			href: {
+				type: String,
+				reflect: true
+			},
+			target: {
+				type: String,
+				reflect: true
+			},
+			text: {
+				type: String,
+				reflect: true
+			},
+			ariaLabel: {
+				attribute: 'aria-label',
+				type: String,
+				reflect: true
+			}
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				align-items: center;
+				display: inline-flex;
+			}
+
+			:host([compact]) .d2l-breadcrumb-wrapper {
+				display: flex;
+				flex-direction: row-reverse;
+				align-items: center;
+			}
+
+			d2l-icon {
+				height: 8px;
+				padding-left: 10px;
+				padding-right: 5px;
+				width: 8px;
+			}
+
+			:host([dir="rtl"]) d2l-icon {
+				padding-left: 5px;
+				padding-right: 10px;
+			}
+
+			:host([compact]) d2l-icon {
+				padding-right: 10px;
+				padding-left: 0;
+			}
+
+			:host([dir="rtl"][compact]) d2l-icon {
+				padding-right: 0;
+				padding-left: 10px;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.href = '#';
```

Comment:
I don't love setting the default value here to `#`... I know that's what it was before but it makes me really sad. I'm guessing this was being done so that the link would get the hand cursor and a tab stop even when it didn't have an `href` set. Curious if that's actually a valid use case...

Reply:
It looks like there are at least some places that use it with `click` event handlers without an `href` so I don't think we can remove it as default unless we also clean those places up.

Reply:
Right. So we can either keep it, or add `tabindex="0"` and `cursor: pointer` when `href` is missing.

Reply:
I will bring it back for now.

Reply:
Actually I didn't remove it in the first place . So I guess: Done ✔️  for now

Reply:
So this isn't great for accessibility. I don't think the default should be `#` when you have an @click. In discover, we use @click since it is an fra and we use single page navigation but our href still equals the actual url that will take you to that page. Prevent default is great for these situations!  

This is what we should make people do for accessibility! Making this default in core says, this is something that should be done when it isn't.

Reply:
@mthjones so just to confirm. removing the default is safe by itself? or should I add the properties Dave mentioned? `(tabindex ="0")` and `cursor: "pointer"`

Reply:
It's not safe by itself, since Allan found some places that are _only_ setting `onclick`, which if we remove this those links will no longer get tab stops.

I think the proper solution here would be to first fix those places that are relying on this and then I agree with Maya that we should remove the default and include some documentation about how to properly provide a `href` AND `click` handler and avoid navigation using `preventDefault`.

I'm still worried that there'll be cases where there literally is no navigation to do. This is why Labs exists -- so that we could make breaking changes like this without impacting places that use the non-core thing... but that ship has sailed for this component.

Reply:
But it's a lot to ask of @miadabrin to go through all those places and fix them up... so I'm not sure how to proceed here.

Reply:
I only found the one case and stopped looking so it might be worth seeing how many occurrences there actually are. To me the two path forwards seem to be:
1. Documenting it as required even though there are technically some places that don't set it
2. Cleaning up places that don't set it and actually making it required but that could prove to be difficult because we'd have to be able to find an appropriate URL to set which may not be obvious.

The former doesn't seem unreasonable to me.

Reply:
So this should be the full list of places that don't set it:

https://search.d2l.dev/xref/GitHub/Brightspace/publish-fra/src/publish-activity-log.js?r=ecd4c66b#134
https://search.d2l.dev/xref/GitHub/Brightspace/publish-fra/src/publish-packages.js?r=a87e11c0#398
https://search.d2l.dev/xref/GitHub/Brightspace/smart-curriculum/src/react/ui/ContentPanel/CreateMaterials.jsx?r=394264f6#212
https://search.d2l.dev/xref/GitHub/Brightspace/smart-curriculum/src/react/ui/ContentPanel/AddMaterials.jsx?r=0ed0dffb#336

If we can figure out what to set for the href (it can just "#" for now I guess) for these places I think we can safely remove the default here (but documenting it like Allan suggested)

Reply:
Cool yeah I like the idea of setting it to `#` but then in the PR asking for help to determine what the `href` should really be. Ensures we're not breaking anything, but let's us move forward.

Reply:
PRs added :
https://github.com/Brightspace/publish-fra/pull/140
https://github.com/Brightspace/smart-curriculum/pull/1055

currently contacting code owners to review it

Reply:
closed: https://github.com/Brightspace/smart-curriculum/pull/1055
(because fork doesn't run the CI)
added: https://github.com/Brightspace/smart-curriculum/pull/1056

Reply:
This is awesome! Thanks for figuring this out :)

Reply:
The default is now removed ✔️